src/App.jsx

```diff
@@ -0,0 +1,18 @@
+import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
+import CityList from "./components/CityList";
+import CityDetails from "./components/CityDetails";
+
+function App() {
+  return (
+    <>
+      <Router>
+        <Routes>
+          <Route path="/city/:index" element={<CityDetails />} />
+          <Route path="/weather-app-react" element={<CityList />} />
+        </Routes>
+      </Router>
+    </>
```

Comment:
There are several issues here:
1. Extra tag.
2. I would suggest to add an empty line between sibling component everywhere.
3. I think having `element` like a JSX looks strange at least. Is it possible to simplify?

```suggestion
      <Router>
        <Routes>
          <Route path="/city/:index" element={CityDetails} />
          
          <Route path="/weather-app-react" element={CityList} />
        </Routes>
      </Router>
```

Reply:
I don't understand about the Extra tag

Reply:
2. +
3. I forgot to remove it after adding the routing, I did.

Reply:
> I don't understand about the Extra tag

I mean this `<> </>`, you don't need this wrapper.